### PR TITLE
refactor: remove business logic, strings from waitlist component

### PR DESCRIPTION
**Pull request by emilyjablonski**
# Pull Request Template

## Issue Overview

This PR addresses #2688

- [x] This change addresses the issue in full

## Description

Removes business logic, hardcoded strings, and styling issues from the Waitlist component. I generalized the name to `QuantityRowSection` so that it can be used outside the context of needing to refer to a waitlist.

## How Can This Be Tested/Reviewed?

[Link to the Storybook deploy preview stories
](https://deploy-preview-2689--dev-storybook-bloom.netlify.app/?path=/story/listing-sidebar-quantity-row-section--open-and-all-values-supplied)
We have a new prop set that adds the strings, and removes the `isWaitlistOpen` check, which we now check before we render the component. Additionally allows JSX content in the description.

The new style updates are: a darker header to meet color contrast, & larger and fully bolded quantity lines.

This is a breaking change to this component due to prop changes.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

**Comment by emilyjablonski** [comment]
Note to self: Breaking Change

File: ui-components/src/page_components/listing/listing_sidebar/QuantityRowSection.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+  return (
+    <section className="aside-block is-tinted">
+      <h4 className="text-caps-tiny">{strings.sectionTitle}</h4>
+      <div>
+        {strings.description && (
+          <p className="text-tiny text-gray-800 pb-3">{strings.description}</p>
+        )}
+        {quantityRows.length && (
+          <ul>{quantityRows.map((row) => (row.amount !== null ? getRow(row) : <></>))}</ul>
```
Comment (seanmalbert) [inline ui-components/src/page_components/listing/listing_sidebar/QuantityRowSection.tsx:41]: If amount is null then React will throw the `Each child in a list should have a unique "key" prop` warning, so we need a key specified on the empty fragment too. That leads me to a question though. If we don't want to display the list item if the amount is null, why allow it to be null at all?

**Review by seanmalbert — CHANGES_REQUESTED** [review]

File: ui-components/src/page_components/listing/listing_sidebar/QuantityRowSection.tsx (hunk 1 above)
Comment (emilyjablonski) [reply ui-components/src/page_components/listing/listing_sidebar/QuantityRowSection.tsx]: Good call, I removed the null check since it's a required field

**Review by seanmalbert — APPROVED** [review]
thanks for the update, lgtm!

**Pull request merged**